summingbird-storm/src/test/scala/com/twitter/summingbird/storm/TestKeyValueSpout.scala

```diff
@@ -0,0 +1,152 @@
+package com.twitter.summingbird.storm
+
+import backtype.storm.spout.{ ISpoutOutputCollector, SpoutOutputCollector }
+import com.twitter.summingbird.storm.spout.{ KeyValueSpout, TraversableSpout }
+import backtype.storm.task.TopologyContext
+import backtype.storm.topology.IRichSpout
+import backtype.storm.tuple.Values
+import com.twitter.algebird.Semigroup
+import com.twitter.algebird.util.summer.{ BufferSize, FlushFrequency, Incrementor, MemoryFlushPercent, SyncSummingQueue }
+import com.twitter.summingbird.online.executor.KeyValueShards
+import com.twitter.summingbird.online.option.SummerBuilder
+import com.twitter.summingbird.storm.spout.KeyValueSpout
+import com.twitter.tormenta.spout.{ BaseSpout, Spout }
+import com.twitter.util.Duration
+import com.twitter.algebird.util.summer.SyncSummingQueue
+import com.twitter.summingbird.batch.{ BatchID, Timestamp }
+import com.twitter.summingbird.storm.collector.AggregatorOutputCollector
+import java.util
+import java.util.HashMap
+import org.junit.runner.RunWith
+import org.scalatest.{ FunSuite, WordSpec }
+import org.scalatest.junit.JUnitRunner
+import java.util.List
+import org.scalacheck._
+import scala.collection.mutable.{ Set => MSet }
+
+/**
+ * Created by pnaramsetti on 8/18/16.
+ */
+
+object TestKeyValueSpout {
+  def getSyncSummingQueueBuildSummer(batchSize: Int, flushFrequency: Int, memFlushPercent: Int) = {
```

Comment:
I have made it to accept a Duration. I am passing the duration in the caller now.